src/main/java/org/jsoup/helper/SpreadsheetGenerator.java

```diff
@@ -0,0 +1,156 @@
+                    if (isMaintainSpan && rowspanList.containsKey(colIndex)) {
+                        int rowspans = rowspanList.get(colIndex);
+                        rowspanList.put(colIndex, --rowspans);
+                        if (rowspans == 0) {
+                            rowspanList.remove(colIndex);
+                        }
+                        continue;
+                    }
+
+                    // check if the cell has colspan or rowspan
+                    if (isMaintainSpan && (cell.getCellColSpan() > 1 || cell.getCellRowSpan() > 1)) {
+                        // merge cells with colspan and rowspan
+                        CellRangeAddress range = new CellRangeAddress(rowIndex, rowIndex + cell.getCellRowSpan() - 1,
+                                colIndex, colIndex + cell.getCellColSpan() - 1);
+                        sheet.addMergedRegion(range);
+
+                        // set the value in the top-left merged cell
+                        Cell firstCell = sheet.getRow(rowIndex).getCell(colIndex);
+                        firstCell.setCellValue(cell.getCellValue());
+
+                        // apply cell style to the merged region
+                        CellStyle cellStyle = workbook.createCellStyle();
+                        cellStyle.setWrapText(true);
+                        sheetCell.setCellStyle(cellStyle);
+
+                        // update rowspanList if rowspan > 1
+                        if (cell.getCellRowSpan() > 1) {
+                            for (int colspanIndex = 0; colspanIndex < cell.getCellColSpan(); colspanIndex++) {
+                                rowspanList.put(colIndex + colspanIndex, cell.getCellRowSpan() - 1);
+                            }
+                        }
+
+                        // skip the columns covered by colspan
+                        if (cell.getCellColSpan() > 1 && cell.getCellRowSpan() == 1) {
+                            colIndex += cell.getCellColSpan() - 1;
+                        }
+                    } else {
+                        // no colspan or rowspan, set cell value
+                        sheetCell.setCellValue(cell.getCellValue());
+
+                        // apply cell style with wrap text
+                        CellStyle cellStyle = workbook.createCellStyle();
+                        cellStyle.setWrapText(true);
+                        sheetCell.setCellStyle(cellStyle);
+                    }
+                }
+            }
+
+            // autosize columns to fit the content
+            for (int colIndex = 0; colIndex < tableWidth; colIndex++) {
+                sheet.autoSizeColumn(colIndex);
+            }
+        }
+
+        // create the file path if it doesn't exist
+        File file = new File(fileSavePath);
+        FileUtils.forceMkdirParent(file);
+        FileUtils.touch(file);
+
+        // write the workbook to the file
+        FileOutputStream outputStream = new FileOutputStream(file);
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**
This line of code might contain a resource leak. Resource leaks can cause your system to slow down or crash.

**Fix**
Consider closing the following resource: *outputStream*. Execution paths that do not contain closure statements were detected. Ensure *outputStream* is closed in a try-finally block or declared in a try-with-resources block.

**More info**
[View resource management guidelines at oracle.com](https://www.oracle.com/technetwork/articles/java/trywithresources-401775.html) (external link).